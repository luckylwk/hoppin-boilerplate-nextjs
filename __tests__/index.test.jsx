import renderer from 'react-test-renderer';
import { HoppinDesignProvider } from 'hoppin-design-system';

import IndexPage from 'pages/index';
import ExperiencePage from 'pages/experience/[slug]';

describe('/', () => {
  it('should mount without failing', () => {
    const component = renderer.create(
      <HoppinDesignProvider context="hopper">
        <IndexPage />
      </HoppinDesignProvider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('/experience/hip-hop-choreographer', () => {
  it('should mount without failing', () => {
    const experience = {
      id: 1,
      slug: 'hip-hop-choreographer',
      title: 'Hip Hop Choreographer',
      location: {},
      hosts_details: {
        profile: {},
      },
      specifics: {},
      price: {},
      information: {
        activities: [],
      },
    };

    const component = renderer.create(
      <HoppinDesignProvider context="hopper">
        <ExperiencePage experience={experience} />,
      </HoppinDesignProvider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
