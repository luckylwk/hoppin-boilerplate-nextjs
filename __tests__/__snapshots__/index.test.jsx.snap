// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ should mount without failing 1`] = `
Array [
  <div>
    TODO: Header
  </div>,
  <div>
    <h3>
      Homepage. Should never be really seen in production.
    </h3>
    <p>
      Check out a
       
      <a
        href="/experience/hip-hop-choreographer"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        sample experience
      </a>
      .
    </p>
    <p>
      This is how a link to 
      <a
        href="https://app.gohoppin.com/profile"
      >
        profile
      </a>
       would work (proxied by 
      <code>
        now
      </code>
      ).
    </p>
  </div>,
  <div>
    TODO: Footer
  </div>,
]
`;

exports[`/experience/hip-hop-choreographer should mount without failing 1`] = `
Array [
  <div>
    TODO: Header
  </div>,
  <div>
    <div
      className="sc-bdVaJa gypVyQ"
      display="block"
      style={
        Object {
          "background": "#f0f4f8",
          "zIndex": "2",
        }
      }
    >
      <div
        className="sc-bdVaJa sc-bwzfXH ieBozk"
        display="flex"
        width="100%"
      >
        <div
          className="sc-bdVaJa eNhJvB"
          display="block"
          order={
            Array [
              2,
              2,
              1,
            ]
          }
          width={
            Array [
              1,
              1,
              "60%",
            ]
          }
        >
          <div
            className="sc-bdVaJa dyfxjG"
            display="block"
          >
            <h1
              className="sc-ifAKCX fYubEk"
            >
              <span>
                Hip Hop Choreographer
              </span>
            </h1>
          </div>
          <div
            className="sc-bdVaJa sc-eNQAEJ fFcKEp"
            display="block"
          >
            <p
              className="sc-EHOje bAKYKb"
              color="inherit"
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <span />
            </p>
          </div>
          <div
            className="sc-bdVaJa feztUM"
            display="block"
          >
            <div
              className="sc-bdVaJa sc-bwzfXH sc-dxgOiQ ebBkek"
              display="flex"
            >
              <div
                className="sc-bdVaJa sc-ckVGcZ hsgOBt"
                display="block"
              >
                <svg
                  fill="currentColor"
                  height={18}
                  size={18}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                      "position": "relative",
                      "top": "0.2em",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.5 3l-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z"
                  />
                </svg>
                <span
                  className="sc-gzVnrw czXKrV"
                  color="inherit"
                  fontFamily="fonts.secondary"
                  fontSize="inherit"
                />
              </div>
              <div
                className="sc-bdVaJa sc-ckVGcZ hsgOBt"
                display="block"
              >
                <svg
                  fill="currentColor"
                  height={18}
                  size={18}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                      "position": "relative",
                      "top": "0.2em",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 17c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1zm0-14v4h2V5.08c3.39.49 6 3.39 6 6.92 0 3.87-3.13 7-7 7s-7-3.13-7-7c0-1.68.59-3.22 1.58-4.42L12 13l1.41-1.41-6.8-6.8v.02C4.42 6.45 3 9.05 3 12c0 4.97 4.02 9 9 9 4.97 0 9-4.03 9-9s-4.03-9-9-9h-1zm7 9c0-.55-.45-1-1-1s-1 .45-1 1 .45 1 1 1 1-.45 1-1zM6 12c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1z"
                  />
                </svg>
                <span
                  className="sc-gzVnrw czXKrV"
                  color="inherit"
                  fontFamily="fonts.secondary"
                  fontSize="inherit"
                />
              </div>
              <div
                className="sc-bdVaJa sc-ckVGcZ hsgOBt"
                display="block"
              >
                <svg
                  fill="none"
                  height={18}
                  size={18}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                      "position": "relative",
                      "top": "0.2em",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.36 6.64a9 9 0 1 1-12.73 0"
                  />
                  <line
                    x1="12"
                    x2="12"
                    y1="2"
                    y2="12"
                  />
                </svg>
                <span
                  className="sc-gzVnrw czXKrV"
                  color="inherit"
                  fontFamily="fonts.secondary"
                  fontSize="inherit"
                >
                  Hands-on activities
                </span>
              </div>
            </div>
            <div
              className="sc-bdVaJa sc-bwzfXH sc-dxgOiQ ebBkek"
              display="flex"
            >
              <div
                className="sc-bdVaJa sc-ckVGcZ hsgOBt"
                display="block"
              >
                <svg
                  fill="currentColor"
                  height={18}
                  size={18}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                      "position": "relative",
                      "top": "0.2em",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
                  />
                </svg>
                <span
                  className="sc-gzVnrw czXKrV"
                  color="inherit"
                  fontFamily="fonts.secondary"
                  fontSize="inherit"
                />
                <span
                  className="sc-gzVnrw gVfdzz"
                  color="inherit"
                  fontFamily="fonts.secondary"
                  fontSize="inherit"
                  fontWeight="bold"
                >
                  Host fee goes to charity
                </span>
              </div>
            </div>
            <div
              className="sc-bdVaJa sc-jKJlTe ctheCo"
              display="block"
            >
              <button
                className="sc-iwsKbI hZXfiI"
                disabled={false}
                display="block"
                onClick={[Function]}
                size="base"
                width="auto"
              >
                Learn more
              </button>
              <button
                className="sc-iwsKbI geQnY"
                disabled={false}
                display="block"
                size="base"
                type="primary"
                width="auto"
              >
                Book your spot now
              </button>
            </div>
          </div>
        </div>
        <div
          className="sc-bdVaJa cOUaTs"
          display="block"
          order={
            Array [
              1,
              1,
              2,
            ]
          }
          width={
            Array [
              1,
              1,
              "40%",
            ]
          }
        >
          <div
            className="sc-bdVaJa sc-kpOJdX kIHWJN"
            display="block"
            src="https://storage.googleapis.com/hoppin-platform/CDN_LIVE/images/hoppin-background.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa sc-jAaTju etSMls"
      display="block"
    >
      <div
        className="sc-bdVaJa sc-bwzfXH fwOzqV"
        display="flex"
      >
        <div
          className="sc-bdVaJa dTLEIn"
          display="block"
          width={1}
        >
          <h5
            className="sc-ifAKCX kdnlcN"
          >
            SIGNATURE EXPERIENCE
          </h5>
          <h2
            className="sc-ifAKCX kSNRAB"
          >
            Hip Hop Choreographer
          </h2>
        </div>
      </div>
      <div
        className="sc-bdVaJa sc-bwzfXH ceEosD"
        display="flex"
      >
        <div
          className="sc-bdVaJa jekwYT"
          display="block"
          order={
            Array [
              1,
              2,
            ]
          }
          width={
            Array [
              1,
              1,
              "40%",
            ]
          }
        >
          <div
            className="sc-bdVaJa sc-jzJRlG hJqWhz"
            display="flex"
            elevation={3}
          >
            <div
              className="sc-bdVaJa gypVyQ"
              display="block"
              style={
                Object {
                  "padding": "24px 24px 0px 24px",
                }
              }
            >
              <div
                className="sc-bdVaJa sc-bwzfXH ifdYuM"
                display="flex"
              >
                <div
                  className="sc-bdVaJa kICUFD"
                  display="block"
                >
                  <h6
                    className="sc-ifAKCX eXBlYt"
                  >
                    your host
                  </h6>
                  <h2
                    className="sc-ifAKCX kSNRAB"
                  >
                    
                     
                    
                  </h2>
                </div>
                <div
                  className="sc-bdVaJa sc-bwzfXH gRBgqm"
                  display="flex"
                >
                  <div
                    className="sc-bdVaJa sc-hMqMXs deFSgX"
                    display="block"
                    size="normal"
                  />
                </div>
              </div>
            </div>
            <div
              className="sc-bdVaJa sc-bwzfXH dyNXwZ"
              display="flex"
            />
            <div
              className="sc-bdVaJa gypVyQ"
              display="block"
              style={
                Object {
                  "padding": "8px 24px 24px 24px",
                }
              }
            />
          </div>
        </div>
        <div
          className="sc-bdVaJa flxAMc"
          display="block"
          order={
            Array [
              2,
              1,
            ]
          }
          width={
            Array [
              1,
              1,
              "60%",
            ]
          }
        >
          <div
            className="sc-bdVaJa sc-jWBwVP yQFlk"
            display="block"
          >
            <div
              className="sc-bdVaJa gypVyQ"
              display="block"
            >
              <h6
                className="sc-ifAKCX eXBlYt"
              >
                What you can expect from the day
              </h6>
            </div>
            <div
              className="sc-bdVaJa gypVyQ"
              display="block"
              style={
                Object {
                  "paddingTop": "48px",
                }
              }
            >
              <h6
                className="sc-ifAKCX eXBlYt"
              >
                WHAT AREAS EXCITE YOU THE MOST FOR THIS EXPERIENCE
              </h6>
              <p
                className="sc-bZQynM hkIQfb"
                color="inherit"
              >
                We always try to personalise your experience as much as we can. What areas excite you the most for this experience? You can choose up to 3:
              </p>
              <div
                className="sc-bdVaJa sc-bwzfXH dKKQjg"
                display="flex"
              />
            </div>
            <div
              className="sc-bdVaJa gypVyQ"
              display="block"
              style={
                Object {
                  "paddingTop": "48px",
                }
              }
            >
              <h6
                className="sc-ifAKCX eXBlYt"
              >
                WHAT YOU NEED TO KNOW
              </h6>
              <div
                className="sc-bdVaJa gypVyQ"
                display="block"
              >
                <table>
                  <tbody>
                    <tr>
                      <td
                        className="sc-brqgnP sc-cMljjf hOTdJq"
                      >
                        <svg
                          fill="currentColor"
                          height={14}
                          size={14}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#627d98",
                            }
                          }
                          viewBox="0 0 24 24"
                          width={14}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11 17c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1zm0-14v4h2V5.08c3.39.49 6 3.39 6 6.92 0 3.87-3.13 7-7 7s-7-3.13-7-7c0-1.68.59-3.22 1.58-4.42L12 13l1.41-1.41-6.8-6.8v.02C4.42 6.45 3 9.05 3 12c0 4.97 4.02 9 9 9 4.97 0 9-4.03 9-9s-4.03-9-9-9h-1zm7 9c0-.55-.45-1-1-1s-1 .45-1 1 .45 1 1 1 1-.45 1-1zM6 12c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1z"
                          />
                        </svg>
                      </td>
                      <td
                        className="sc-brqgnP cdpvSD"
                      >
                        The experience takes
                         
                        <span
                          className="sc-gzVnrw izswwR"
                          color="inherit"
                          fontFamily="fonts.secondary"
                          fontSize="inherit"
                          fontWeight="bold"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="sc-brqgnP sc-cMljjf hOTdJq"
                      >
                        <svg
                          fill="none"
                          height={14}
                          size={14}
                          stroke="currentColor"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          style={
                            Object {
                              "color": "#627d98",
                            }
                          }
                          viewBox="0 0 24 24"
                          width={14}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="12"
                            cy="12"
                            r="10"
                          />
                        </svg>
                      </td>
                      <td
                        className="sc-brqgnP cdpvSD"
                      >
                        Hoppers are
                         
                        <span
                          className="sc-gzVnrw izswwR"
                          color="inherit"
                          fontFamily="fonts.secondary"
                          fontSize="inherit"
                          fontWeight="bold"
                        >
                          allowed to blog, post, tweet
                        </span>
                         on Social Media
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="sc-brqgnP sc-cMljjf hOTdJq"
                      >
                        <svg
                          fill="currentColor"
                          height={14}
                          size={14}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#627d98",
                            }
                          }
                          viewBox="0 0 24 24"
                          width={14}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M20.5 3l-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z"
                          />
                        </svg>
                      </td>
                      <td
                        className="sc-brqgnP cdpvSD"
                      >
                        Experience location will be shared by the host
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="sc-brqgnP sc-cMljjf hOTdJq"
                      >
                        <svg
                          fill="currentColor"
                          height={14}
                          size={14}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#627d98",
                            }
                          }
                          viewBox="0 0 24 24"
                          width={14}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
                          />
                        </svg>
                      </td>
                      <td
                        className="sc-brqgnP cdpvSD"
                      >
                        The price is
                         
                        <span
                          className="sc-gzVnrw izswwR"
                          color="inherit"
                          fontFamily="fonts.secondary"
                          fontSize="inherit"
                          fontWeight="bold"
                        >
                           
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                className="sc-bdVaJa fsKNXY"
                display="block"
              >
                <button
                  className="sc-iwsKbI hBxfIl"
                  disabled={true}
                  display="block"
                  size="base"
                  type="primary"
                  width="100%"
                >
                  Book your spot now
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    TODO: Footer
  </div>,
  ",",
]
`;
